File: Structural/Lab9/Typescript/src/Entities/estate_container.ts
import { IPrintEstateInfo } from '../Interfaces/print_estate_info';

export class EstateContainer implements IPrintEstateInfo {
    protected _realEstateList: IPrintEstateInfo[] = [];

    public add(estate: IPrintEstateInfo): void {
        this._realEstateList.push(estate);
    }

    public remove(estate: IPrintEstateInfo): void {
        const index = this._realEstateList.indexOf(estate);
        if (index > -1) {
            this._realEstateList.splice(index, 1);
        }
    }

    public printEstateInfo(tab: string = ""): void {
        console.log(tab + "Estate list in container " + ":(");
        this._realEstateList.forEach(estate => {
            estate.printEstateInfo(tab + tab);
        });
        console.log(tab + ")");
    }

}

File: Structural/Lab11/C#/Entities/EstateContainer.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Interfaces;

namespace C_.Entities
{
    public class EstateContainer : IPrintEstateInfo
    {
        protected List<IPrintEstateInfo> _realEstateList = new List<IPrintEstateInfo>();

        public void Add(IPrintEstateInfo estate){
            _realEstateList.Add(estate);
        }
        public void Remove(IPrintEstateInfo estate){
            _realEstateList.Remove(estate);
        }

        public void PrintEstateInfo(string tab = ""){
            Console.WriteLine(tab + "Estate list in container with hash code " + GetHashCode() + ":(");
            foreach(IPrintEstateInfo estate in _realEstateList){
                estate.PrintEstateInfo(tab + tab);
            }
            Console.WriteLine(tab + ")");
        }
    }
}
